pkg/storaged/crypto-keyslots.jsx

```diff
@@ -253,6 +256,191 @@
+                progress(null, null);
+                if (data.missing_names.length + data.unavailable_names.length > 0)
+                    steps.push({
+                        title: cockpit.format(_("The $0 package must be installed."), package_name),
+                        func: progress => {
+                            if (data.remove_names.length > 0)
+                                return Promise.reject(cockpit.format(_("Installing $0 would remove $1."), name, data.remove_names[0]));
+                            else if (data.unavailable_names.length > 0)
+                                return Promise.reject(cockpit.format(_("The $0 package is not available from any repository."), name));
+                            else
+                                return install_missing_packages(data, status_callback(progress));
+                        }
+                    });
+            });
+}
+
+function ensure_initrd_clevis_support(steps, progress, package_name) {
+    const task = cockpit.spawn(["lsinitrd", "-m"], { superuser: true, err: "message" });
+    progress(_("Checking for NBDE support in the initrd"), () => task.close());
+    return task.then(data => {
+        progress(null, null);
+        if (data.indexOf("clevis") < 0) {
+            return ensure_package_installed(steps, progress, package_name)
+                    .then(() => {
+                        steps.push({
+                            title: _("The initrd must be regenerated."),
+                            func: progress => {
+                                // dracut doesn't react to SIGINT, so let's not enable our Cancel button
+                                progress(_("Regenerating initrd"), null);
+                                return cockpit.spawn(["dracut", "--force", "--regenerate-all"],
+                                                     { superuser: true, err: "message" });
+                            }
+                        });
+                    });
+        }
+    });
+}
+
+function ensure_root_nbde_support(steps, progress) {
+    progress(_("Adding rd.neednet=1 to kernel command line"), null);
+    return cockpit.spawn(["grubby", "--update-kernel=ALL", "--args=rd.neednet=1"],
+                         { superuser: true, err: "message" })
+            .then(() => ensure_initrd_clevis_support(steps, progress, "clevis-dracut"));
+}
+
+function ensure_fstab_option(steps, progress, client, block, option) {
+    const cleartext = client.blocks_cleartext[block.path];
+    const crypto = client.blocks_crypto[block.path];
+    const fsys_config = (cleartext
+        ? array_find(cleartext.Configuration, function (c) { return c[0] == "fstab" })
+        : array_find(crypto.ChildConfiguration, function (c) { return c[0] == "fstab" }));
+    const fsys_options = fsys_config && parse_options(decode_filename(fsys_config[1].opts.v));
+
+    if (!fsys_options || fsys_options.indexOf(option) >= 0)
+        return Promise.resolve();
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17796-20221114-133844-c33e9cff-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)